### PR TITLE
[#94618] Homepage fixes

### PR DESCRIPTION
**Pull request by jhanggi**
* Return facilities list to alphabetical order
* Use acting_user as source for most recent facilities
* Order recent facilities by name
* Ensure orders are searched by ordered_at for use in recent facilities
* Add index for the homepage ordering

### CODE REVIEW

File: spec/factories/facilities.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 FactoryGirl.define do
   factory :facility do
-    sequence(:name) { |n| "Facility#{n}" }
+    sequence(:name, "AAAAAAAA") { |n| "Facility#{n}" }
```
Comment (drench) [inline spec/factories/facilities.rb]: We're doing this enough that I wish we had something like `alphabetical_sequence(:name)`.

**Comment by drench** [comment]
:+1:

**Comment by mlineen** [comment]
Why not use `{ |n| "Facility#{n.to_s.rjust(10,'0')}" }`

**Comment by jhanggi** [comment]
I think that's uglier to read, but we could use `sequence(:name, "00000000001")`, which does lead to easier to grok names while debugging.

**Comment by jhanggi** [comment]
And I was playing around, and using that as the starting point, I have an idea for a PR for FactoryGirl to natively (or via monkey patch) support `alphabetic_sequences`.

**Comment by jhanggi** [comment]
Not quite ready for PR: https://github.com/tablexi/nucore-open/compare/master...alphabetic_sequence

**Comment by mlineen** [comment]
Does `sequence(:name, "Facility_0001")` without a block effectively do what we want? 
```
> 'Facility_0001'.next
 => "Facility_0002"
```

**Comment by jhanggi** [comment]
Unfortunately, not.

```
> "Facility_9999".next
 => "Facility_10000"
```

The only thing I can think of that might work is a custom iterator that adds the extra character on the right of the string, e.g.:
```
8, 9, 90, 91, .., 98, 99, 990
A, B, ..., Y, Z, ZA, .., ZY, ZZ, ZZA
```

I think that's almost more confusing if you're in debugging, especially in the numerical version.

I'm open to other suggestions. This has come up multiple times across multiple projects in the past couple weeks, so I do have a vested interest in finding the right solution.

**Comment by mlineen** [comment]
I guess what I was advocating is in tests using enough leading zeros so that you never run into the alphabetical issue which is basically what your code is doing, but I'm not creating the safety net of warning if you wrap.

**Pull request merged**